### DISCUSSION TITLE
More Dask support throughout

### DISCUSSION THREAD

**Discussion by kenkehoe** [ARM-DOE/ACT#353] [General]
Should we refactor our current code to utilize Dask more?

**Comment by AdamTheisen** [comment]
What do you have in mind?  For some of the more heavy compute items, we're already using dask to distribute processing.

io/mpl.py:        task.append(dask.delayed(proc_sigma_mplv5_read)(f, save_nc=save_nc,
qc/radiometer_tests.py:        task.append(dask.delayed(fft_shading_test_process)(
retrievals/irt.py:        task.append(dask.delayed(process_sst_data)(sfc_temp[i], sky_temp[i], emis,
utils/ship_utils.py:        task.append(dask.delayed(proc_scog)

**Reply by kenkehoe** [reply]
I guess we could go through and find more places to add Dask where it is not currently utilized. Are there places where things are slower than they should be?

**Reply by AdamTheisen** [reply]
Good point and I think there are places we could utilize it but we have to just test.  Sometime's it's not faster than just a regular for loop.  Just depends on the amount of data and what needs to be done. I think it's worthwhile to investigate though
